core/src/androidTest/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/YoutubePlayerViewDefaultWebUITest.kt

```diff
@@ -0,0 +1,170 @@
+package com.pierfrancescosoffritti.androidyoutubeplayer.core
+
+import android.app.Activity
+import android.view.View
+import android.widget.FrameLayout
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.filters.LargeTest
+import androidx.test.rule.ActivityTestRule
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.listeners.AbstractYouTubePlayerListener
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.options.IFramePlayerOptions
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.views.LegacyYouTubePlayerView
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.views.YouTubePlayerView
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.testActivity.TestActivity
+import com.pierfrancescosoffritti.androidyoutubeplayer.test.R
+import org.hamcrest.CoreMatchers.*
+import org.hamcrest.Matcher
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+/**
+ * This class tests if the player's default ui is the web-based ui.
+ * Additionally, it tests all 4 manual initialization methods to check
+ * if they create their expected UI elements.
+ */
+@LargeTest
+class YoutubePlayerViewDefaultWebUITest {
+
+    private lateinit var testActivity: Activity
+    private lateinit var youTubePlayerViewUsingInitializer1: YouTubePlayerView
+    private lateinit var youTubePlayerViewUsingInitializer2: YouTubePlayerView
+    private lateinit var youTubePlayerViewUsingInitializer3: YouTubePlayerView
+    private lateinit var youTubePlayerViewUsingInitializeWithNativeUi: YouTubePlayerView
+
+    @Rule
+    @JvmField
+    val activityRule = ActivityTestRule(TestActivity::class.java)
+
+    @Before
+    fun setup() {
+        testActivity = activityRule.activity
+        youTubePlayerViewUsingInitializer1 = testActivity.findViewById(R.id.youtube_player_view_manual_initialize_1)
+        youTubePlayerViewUsingInitializer2 = testActivity.findViewById(R.id.youtube_player_view_manual_initialize_2)
+        youTubePlayerViewUsingInitializer3 = testActivity.findViewById(R.id.youtube_player_view_manual_initialize_3)
```

Comment:
It's not clear how youTubePlayerViewUsingInitializer1, youTubePlayerViewUsingInitializer2 and youTubePlayerViewUsingInitializer3 differ from each other. Can you leave a comment?

Reply:
done